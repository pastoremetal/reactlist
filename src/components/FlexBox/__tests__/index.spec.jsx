import React from 'react';
import { render } from 'enzyme';
import { Flex, Box } from '../index';

describe('Flex', () => {
  it('should be a div for default', () => {
    const wrapper = render(<Flex />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be a given component', () => {
    const child = () => <header />;
    const wrapper = render(<Flex tag={child} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain a given child component', () => {
    const wrapper = render(<Flex><header>foo</header></Flex>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Box', () => {
  it('should be a div for default', () => {
    const wrapper = render(<Box />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be a given component', () => {
    const child = () => <header />;
    const wrapper = render(<Box tag={child} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain a given child component', () => {
    const wrapper = render(<Box><header>foo</header></Box>);
    expect(wrapper).toMatchSnapshot();
  });
});
