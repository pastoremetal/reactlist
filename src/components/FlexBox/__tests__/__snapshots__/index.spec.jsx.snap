// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should be a div for default 1`] = `
<div
  class="sc-bdVaJa cnkOUi"
/>
`;

exports[`Box should be a given component 1`] = `<header />`;

exports[`Box should contain a given child component 1`] = `
<div
  class="sc-bdVaJa cnkOUi"
>
  <header>
    foo
  </header>
</div>
`;

exports[`Flex should be a div for default 1`] = `
<div
  class="sc-bwzfXH jIEWCi"
/>
`;

exports[`Flex should be a given component 1`] = `<header />`;

exports[`Flex should contain a given child component 1`] = `
<div
  class="sc-bwzfXH jIEWCi"
>
  <header>
    foo
  </header>
</div>
`;
